api/src/test/java/com/epam/pipeline/acl/cluster/ClusterApiServiceTest.java

```diff
@@ -0,0 +1,546 @@
+import com.epam.pipeline.security.acl.AclPermission;
+import com.epam.pipeline.test.acl.AbstractAclTest;
+import com.epam.pipeline.test.creator.cluster.NodeCreatorUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import java.io.InputStream;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doReturn;
+
+public class ClusterApiServiceTest extends AbstractAclTest {
+
+    @Autowired
+    private ClusterApiService clusterApiService;
+
+    @Autowired
+    private NodesManager mockNodesManager;
+
+    @Autowired
+    private PipelineRunManager mockPipelineRunManager;
+
+    @Autowired
+    private ContextualPreferenceManager mockContextualPreferenceManager;
+
+    @Autowired
+    private AuthManager mockAuthManager;
+
+    @Autowired
+    private UsageMonitoringManager mockUsageMonitoringManager;
+
+    @Autowired
+    private NodeDiskManager mockNodeDiskManager;
+
+    @Autowired
+    private InstanceOfferManager mockInstanceOfferManager;
+
+    private final FilterPodsRequest filterPodsRequest = NodeCreatorUtils.getDefaultFilterPodsRequest();
+
+    private final NodeInstance nodeInstance = NodeCreatorUtils.getDefaultNodeInstance();
+
+    private final NodeInstance nodeInstanceWithoutPermission = NodeCreatorUtils.getDefaultNodeInstance();
+
+    private final PipelineRun pipelineRun = NodeCreatorUtils.getPipelineRun();
+
+    private final ContextualPreference contextualPreference = NodeCreatorUtils.getContextualPreference();
+
+    private final FilterNodesVO filterNodesVO = NodeCreatorUtils.getDefaultFilterNodesVO();
+
+    private final NodeDisk nodeDisk = NodeCreatorUtils.getDefaultNodeDisk();
+
+    private final MonitoringStats monitoringStats = NodeCreatorUtils.getMonitoringStats();
+
+    private final InstanceType instanceType = NodeCreatorUtils.getDefaultInstanceType();
+
+    private InputStream inputStream;
+
+    private List<NodeDisk> nodeDisks;
+
+    private List<InstanceType> instanceTypes;
+
+    private List<MonitoringStats> statsList;
+
+    private List<NodeInstance> singleNodeInstance;
+
+    private List<NodeInstance> twoNodeInstances;
+
+    @Before
+    public void setUp() {
+        statsList = Collections.singletonList(monitoringStats);
+
+        instanceTypes = Collections.singletonList(instanceType);
+
+        nodeDisks = Collections.singletonList(nodeDisk);
+
+        pipelineRun.setId(1L);
+        pipelineRun.setOwner(SIMPLE_USER);
+
+        final PipelineRun pipelineRunWithoutPermission = NodeCreatorUtils.getPipelineRun();
```

Comment:
Let's create a new build method in creators package. This  method may accept `id`, `name`, `owner` arguments.